### PR TITLE
[docs] minor fixes to CONTRIBUTING.rst

### PR DESCRIPTION
**Pull request by jameslamb**
I've been using `CONTRIBUTING.rst` while working on #149 . Thanks for maintaining such thorough docs for contributors!

This PR proposes some minor cleanup to issues I saw in there

1. Use of `./tox` should just be `tox`, to avoid the error `-bash: ./tox: No such file or directory`
2. Minor typo fixes
3. Replace some `::` with `:`

### CODE REVIEW

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -29,19 +29,19 @@ Non-Goals:
 Development
 -----------
 
-For many tasks, it is okay to just develop using a single installed python version. But if you need to test/debug the project in multiple python versions, you need to install those version::
```
Comment (tkruse) [inline CONTRIBUTING.rst:32]: The duplicate colon is not a typo in reStructuredText, e.g. see https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#literal-blocks.

Comment (jameslamb) [reply CONTRIBUTING.rst:32]: ah sorry! I didn't know about that. Will revert them.

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -29,19 +29,19 @@ Non-Goals:
...
+For many tasks, it is okay to just develop using a single installed python version. But if you need to test/debug the project in multiple python versions, you need to install those version:
```
Comment (jameslamb) [inline CONTRIBUTING.rst:32]:
```suggestion
For many tasks, it is okay to just develop using a single installed python version. But if you need to test/debug the project in multiple python versions, you need to install those version::
```

File: CONTRIBUTING.rst [hunk 3]
```diff
@@ -29,19 +29,19 @@ Non-Goals:
...
 
 1. (Optional) Install multiple python versions
 
    1. (Optional) Install [pyenv](https://github.com/pyenv/pyenv-installer) to manage python versions
-   2. (Optional) Using pyenv, install the python versions used in testing::
+   2. (Optional) Using pyenv, install the python versions used in testing:
```
Comment (jameslamb) [inline CONTRIBUTING.rst:37]:
```suggestion
   2. (Optional) Using pyenv, install the python versions used in testing::
```

File: CONTRIBUTING.rst [hunk 4]
```diff
@@ -29,19 +29,19 @@ Non-Goals:
...
 
         pyenv install 2.7.16
         pyenv install 3.6.8
 
 It may be okay to run and test python against locally installed libraries, but if you need to have a consistent build, it is recommended to manage your environment using virtualenv: [virtualenv](https://virtualenv.pypa.io/en/latest/ ), [virtualenvwrapper](https://pypi.org/project/virtualenvwrapper/ ):
 
-1. (Optional) Setup a local virtual environment with all necessary tools and libraries::
+1. (Optional) Setup a local virtual environment with all necessary tools and libraries:
```
Comment (jameslamb) [inline CONTRIBUTING.rst:44]:
```suggestion
1. (Optional) Setup a local virtual environment with all necessary tools and libraries::
```

**Pull request merged**